### DISCUSSION TITLE
How can I remove the unwanted div.wp-block-template-part wrapper from Template Part

### DISCUSSION THREAD

**Discussion by realthemes** [WordPress/gutenberg#54950] [Q&A] [answered]
Hello, how can I remove the unwanted `div.wp-block-template-part` wrapper from Template Part that wraps my custom navigation block on the frontend? This additional div wrapper is disrupting the sticky behavior of my navigation.
I want my custom navigation block to behave like a Template Part but not generate an additional div wrapper.

**Comment by patrickreiner** [comment]
Agreed, that extra div is so annoying

**Comment by huubl** [comment, marked as answer]
I think there is [no proper way](https://github.com/WordPress/gutenberg/issues/49071) to remove the wrapper from template parts. But I came across [this repository](https://github.com/alleyinteractive/create-wordpress-theme) that uses workaround to allow template parts to render [without a wrapper](https://github.com/alleyinteractive/create-wordpress-theme/blob/develop/inc%2Fblocks.php#L13-L54) when a custom attribute ["skipWrapper"](https://github.com/alleyinteractive/create-wordpress-theme/blob/develop/templates%2Findex.html) is set. While I haven’t personally tested this, it might be of assistance to you.

**Reply by patrickreiner** [reply]
Wow! In that repository, the function remove_core_template_part_wrapper in [blocks.php](https://github.com/alleyinteractive/create-wordpress-theme/blob/develop/inc/blocks.php) does exactly that.

This is perfect, thank you so much.

**Reply by dc774** [reply]
I thought I'd finally found some help with this issue, but that function doesn't seem to exist anymore in blocks.php. I searched the whole repo and couldn't find it. Do you still use it and can you share it?

**Reply by realthemes** [reply]
@dc774 You can view this file in History tab:
<img width="1501" height="268" alt="Screenshot 2025-07-10 181041" src="https://github.com/user-attachments/assets/96650981-8095-4ae7-9cf0-45044b33ca20" />

**Comment by coreyworrell** [comment]
With the addition of Custom CSS per block in WordPress 7.0, you can select the template part in editor and add `display: contents;` there to achieve similar result.
